### PR TITLE
Refactor coversheet to use generic court details key

### PR DESCRIPTION
**Pull request by johnoneill46**
https://tools.hmcts.net/jira/browse/FR-1811

Refactor bulkprint coversheet to use single object for court contact details instead of using hard coded values in Docmosis template.

- Renamed from 'ctscContactDetails' to 'courtContactDetails' as we're introducing FRC logic soon so should be generic to work for both case types
- Similar to how we use Addressee.formattedAddress object

### CODE REVIEW

**Review by ocarleton — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/GenerateCoverSheetService.java [hunk 1]
```diff
@@ -99,11 +105,39 @@ private void prepareCoverSheet(CaseDetails caseDetails, String partyAddressCcdFi
...
             caseData.put(ADDRESSEE, addressee);
-            caseData.put(CTSC_CONTACT_DETAILS, buildCtscContactDetails());
+            caseData.put(COURT_CONTACT_DETAILS, getCourtContactDetails(caseDetails));
             caseData.put(CASE_NUMBER, nullToEmpty(caseDetails.getId()));
         }
     }
 
+    private String getCourtContactDetails(CaseDetails caseDetails) {
+        /*
+        if (isContestedApplication(caseDetails)) {
```
Comment (shaunganley) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/GenerateCoverSheetService.java:115]: this not ready?

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/GenerateCoverSheetService.java [hunk 2]
```diff
@@ -99,11 +105,39 @@ private void prepareCoverSheet(CaseDetails caseDetails, String partyAddressCcdFi
...
+            .build();
+
+        return Stream.of(coversheetCtscContactDetails.getServiceCentre(),
+                coversheetCtscContactDetails.getPoBox(),
+                coversheetCtscContactDetails.getTown(),
+                coversheetCtscContactDetails.getPostcode())
+            .filter(Objects::nonNull)
+            .map(Object::toString)
+            .filter(StringUtils::isNotEmpty)
+            .filter(s -> !s.equals("null"))
```
Comment (shaunganley) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/GenerateCoverSheetService.java:137]: I would probably avoid doing this check, we already have the null check above, if they type null into a text box then fair play to them

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/GenerateCoverSheetServiceTest.java [hunk 3]
```diff
@@ -182,8 +183,14 @@ private void assertCoversheetCalledWithRequiredData() {
...
 
+        String expectedCourtContactDetails = "HMCTS Financial Remedy" + "\n"
+            + "PO BOX 12746" + "\n"
+            + "HARLOW" + "\n"
+            + "CM20 9QZ";
+
         assertThat(data, hasKey(ADDRESSEE));
-        assertThat(data, hasKey(CTSC_CONTACT_DETAILS));
+        assertThat(data, hasKey(COURT_CONTACT_DETAILS));
+        assertEquals(expectedCourtContactDetails, data.get(COURT_CONTACT_DETAILS));
```
Comment (shaunganley) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/GenerateCoverSheetServiceTest.java:193]: can we add it tests for the other operator checks we do i.e. null and empty string

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/GenerateCoverSheetService.java (hunk 1 above)
Comment (johnoneill46) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/GenerateCoverSheetService.java:115]: Not yet - will be doing it as a separate PR.
I want to make sure the consented journey is unaffected when my PR comes in. This way we can use a single method for populating both consented or contested court data on the template

**Review by HarryH96 — APPROVED** [review]

**Pull request merged**